scheduler/scheduler.go

```diff
@@ -281,13 +291,32 @@ func generateContainerHostIntents(d distro.Distro, newContainersNeeded int) ([]h
 	return containerHostIntents, nil
 }
 
+func getDockerOptionsFromProviderSettings(settings *map[string]interface{}) (*host.DockerOptions, error) {
+	dockerOptions := &host.DockerOptions{}
+	if settings != nil {
+		if err := mapstructure.Decode(settings, dockerOptions); err != nil {
+			return nil, errors.Wrap(err, "Error decoding params")
+		}
+	}
+	if dockerOptions.Image == "" {
+		return nil, errors.New("docker image cannot be empty")
+	}
+	return dockerOptions, nil
+}
+
 // generateParentHostOptions generates host options for a parent host
-func generateParentHostOptions(pool *evergreen.ContainerPool) cloud.HostOptions {
-	return cloud.HostOptions{
+func generateParentHostOptions(settings *map[string]interface{}, pool *evergreen.ContainerPool) (cloud.HostOptions, error) {
```

Comment:
pointers to maps are really scary? is there a way to avoid this?

Reply:
ProviderSettings in hosts is already a pointer to a map. I can just pass the map instead though, decode doesn't require a pointer.